### ISSUE TITLE
rasa core state changes and slot filling not intelligently cohesive

### ISSUE THREAD

**Issue by anshulsolanki** [RasaHQ/rasa_core#203]
**Rasa Core version**:
7.9

**Python version**: 
3.6

**Operating system** (windows, osx, ...):
windows

**Issue**:
rasa-nlu identifies the intent and entities, rasa core fills the slots and takes the next action. 

but rasa core performs in a way, where next action and filling of slots is not context aware and both seems to happen in isolation. Here is an example from restaurant bot.

you - Hello 
bot - How can i help you?
you - i need some food.
bot - what kind of cuisine would you like?
you - for six people

at this point the slot for people gets filled by value two. Why it happens. Because NLU said there is entity for 'people' and slot for filled for it. Even though the original question from bot was about cuisine.
Worst, the state machine thinks it has successfully passed that state and never returns to ask for cuisine again.
I my view the answer by user should be read in context of the question asked.

What are some of the thoughts for this? how should this be handled.

**Content of domain file** (if used & relevant):
```yaml

```

**Comment by tmbo** [comment]
Mhm but isn't that (partially) correct? So I'd expect the bot to store that information even if it didn't ask for it. Nevertheless, the bot should, of course, ask for the cuisine again as it didn't get that information.

**Comment by anshulsolanki** [comment]
I feel in order to solve this
1) There needs to be connection between <current utterance by bot> and <slot to be filled >
2) if entity extracted does not match the <slot to be filled > , bot should take the information , respond back to user which information it got
3) and then ask the original question , i.e. <current utterance by bot>  again.

But this should be handled at core not at application level.

@tmbo  what do you suggest

**Comment by tmbo** [comment]
Yes I agree, this is something core could do. The question is how. It is a machine learning model so that information somehow needs to be generated during training and while using the model
